src/features/filter.ts

```diff
@@ -1,10 +1,49 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetQueryAction = { type: 'query/SET', payload: string };
+type RemoveQueryAction = { type: 'query/REMOVE' };
+type SetStatusAction = { type: 'status/SET', payload: Status };
+
+type Action = SetQueryAction | RemoveQueryAction | SetStatusAction;
+
+const setQuery = (query: string): SetQueryAction => (
+  { type: 'query/SET', payload: query }
+);
+const removeQuery = (): RemoveQueryAction => ({ type: 'query/REMOVE' });
+
+const setStatus = (status: Status): SetStatusAction => (
+  { type: 'status/SET', payload: status }
+);
+
+export const actions = { setQuery, removeQuery, setStatus };
+
+type Filter = {
+  query: string,
+  status: Status,
+};
+
+const initiialFilter: Filter = {
```

Comment:
There is a typo in the variable name 'initiialFilter'. It should be 'initialFilter'